aodncore/pipeline/handlerbase.py

```diff
@@ -527,17 +537,14 @@ def _notify_common(self, notify_list_param):
         notify_runner = get_notify_runner(notification_data, self.config, self.logger, self.notify_params)
         self.logger.sysinfo("get_notify_runner -> '{runner}'".format(runner=notify_runner.__class__.__name__))
 
-        notify_params_dict = self.notify_params or {}
-        notify_list = notify_params_dict.get(notify_list_param)
-
-        if notify_list:
-            self._notify_list = notify_runner.run(notify_list)
+        if self._should_notify:
+            self._notification_results = notify_runner.run(self._should_notify)
 
     def _notify_success(self):
-        self._notify_common('success_notify_list')
+        self._notify_common()
```

Comment:
Is it still worth having separate `_notify_success` and `_notify_error` methods?

Reply:
I think so, because they are both associated with a different state transition, so to me it makes sense to be able to potentially act differently for a different transition (even though they are currently just running the same other method!). 

It gives it a 1->1->1 relationship between triggering the state, running it's dedicated method and transitioning to that state, without sharing any "before transition" methods, so keeps it cleaner conceptually.

e.g.
```python
self.state
'HANDLER_POSTPROCESSED'

self.trigger('_trigger_notify_error')  # runs self._notify_error()

self.state
'HANDLER_NOTIFIED_ERROR'
```

Reply:
:+1: Makes sense.